drivers/disk/flashdisk.c

```diff
@@ -397,7 +405,7 @@ DT_INST_FOREACH_STATUS_OKAY(DEFINE_FLASHDISKS_CACHE)
 	.area_id = DT_FIXED_PARTITION_ID(PARTITION_PHANDLE(n)),			\
 	.offset = DT_REG_ADDR(PARTITION_PHANDLE(n)),				\
 	.cache = flashdisk##n##_cache,						\
-	.cache_size = DT_INST_PROP(n, cache_size),				\
+	.cache_size = ARRAY_SIZE(flashdisk##n##_cache),				\
```

Comment:
Why is it not `sizeof(flashdisk##n##_cache),` ?

Reply:
Is there any difference in the value returned by `ARRAY_SIZE()` and `sizeof()`?

Reply:
The difference in in code readability/understanding.

Reply:
Change to sizeof() should be done in the first commit, not in the second.